### PR TITLE
Chore: update macro docs to include runtime_stage macro

### PR DESCRIPTION
**Pull request by GeorgeSittas**

### CODE REVIEW

File: docs/concepts/macros/sqlmesh_macros.md [hunk 1]
```diff
@@ -318,6 +328,21 @@ SELECT
...
 ```
 
+Another example is conditionally executing a pre/post-statement depending on the current [runtime stage](./macro_variables.md#predefined-variables). For instance, the following `@IF` pre-statement will only be executed at model creation time:
+
+```sql linenums="1"
+MODEL (
+  ...
+);
+
+@IF(
```
Comment (tobymao) [inline docs/concepts/macros/sqlmesh_macros.md:338]: have you tested this? i think you want this as a post statement and not a pre statement

Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md:338]: not yet, i can try to test this specific example though soon

Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md:338]: to be clear, i did test using both `create schema ...` and `create table ...` as pre-statements, seemed good

Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md:338]: yeah what you're saying makes sense, I'll try to refine the example

Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md:352]: updated the example to use an evaluation-time post-statement instead

File: docs/concepts/macros/sqlmesh_macros.md [hunk 2]
```diff
@@ -260,10 +260,22 @@ It includes three elements:
...
 
 The value to return if the condition is `FALSE` is optional - if it is not provided and the condition is `FALSE`, then the macro has no effect on the resulting query.
 
+It's also possible to use this macro in order to conditionally execute pre/post-statements:
+
+```sql linenums="1"
+@IF([logical condition], [statement to execute if TRUE])
+```
+
+Note that the statement to be executed must not end with a semi-colon.
```
Comment (treysp) [inline docs/concepts/macros/sqlmesh_macros.md:277]: Looking at the example below, is it correct that the `@IF()` statement SHOULD end with a semicolon but the statement within should not?

File: docs/concepts/macros/sqlmesh_macros.md [hunk 3]
```diff
@@ -260,10 +260,22 @@ It includes three elements:
...
-These elements are specified as `@IF([logical condition], [value if TRUE], [value if FALSE])`.
+These elements are specified as:
+
+```sql linenums="1"
+@IF([logical condition], [value if TRUE], [value if FALSE])
+```
 
 The value to return if the condition is `FALSE` is optional - if it is not provided and the condition is `FALSE`, then the macro has no effect on the resulting query.
 
+It's also possible to use this macro in order to conditionally execute pre/post-statements:
```
Comment (treysp) [inline docs/concepts/macros/sqlmesh_macros.md:271]: Is it possible to have both unconditional and conditional pre/post statements? Like if I have one pre-statement that should always run and a second pre-statement that should only run sometimes?

File: docs/concepts/macros/sqlmesh_macros.md (hunk 2 above)
Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md:277]: That's correct, I can update this note if it doesn't seem clear

File: docs/concepts/macros/sqlmesh_macros.md (hunk 3 above)
Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md:271]: Yeah that's correct. All pre/post-statements are always rendered, but the ones wrapped in `@IF` are simply rendered to `None` IIRC and so they won't included in the execution? I'll need to double check that.

File: docs/concepts/macros/sqlmesh_macros.md (hunk 2 above)
Comment (GeorgeSittas) [reply docs/concepts/macros/sqlmesh_macros.md]: Updated

**Review by treysp — APPROVED** [review]

**Review by tobymao — APPROVED** [review]

**Pull request merged**